portfolio/src/main/webapp/script.js

```diff
@@ -99,13 +99,240 @@ function getComments() {
+      .appendChild(document.createTextNode("You clicked on ")).parentElement
+      .appendChild(countryDOM.parentElement).parentElement);
+}
+
+function noCountriesLeftDOM() {
+
+  document.getElementById("rand_country")
+    .appendChild(document.createElement("h4")
+      .appendChild(document.createTextNode("There are no more contries left!")).parentElement);
+
+  document.getElementById("selected_country")
+    .appendChild(document.createElement("h3")
+      .appendChild(document.createTextNode("Congratulations. You Win!!!")).parentElement);
+
+  document.getElementById("skip_btn").remove();
+}
+
 function initMap() {
+
   const map = new google.maps.Map(
-      document.getElementById('map'),
-      {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+    document.getElementById('map'), {
+    center: { lat: 0, lng: 0 },
+    zoom: 4,
+    disableDefaultUI: true,
+    zoomControl: true,
+    mapTypeControl: false,
+    scaleControl: true,
+    streetViewControl: false,
+    rotateControl: true,
+    fullscreenControl: false,
+    styles: [
+      {
+        featureType: "all",
+        elementType: "labels",
+        stylers: [
+          { visibility: "off" }
+        ]
+      },
+
+      {
+        featureType: "road",
+        stylers: [
+          { visibility: "off" }
+        ]
+      }
+    ]
+  });
+
+  fetch("/documents/countries.json").then(resp => resp.json()).then(countriesJsonFile => {
+
+    let countries = new Countries(countriesJsonFile);
+    let randCountryCodeStr = createRandCountryDOM(countries);
+    let incorrectMarkersArray = [];
+
+    document.getElementById("skip_btn").addEventListener("click", function() {
+
+      if (countries.empty()) {
+        deleteElementContentsById("rand_country");
+        deleteAnswerDOM();
+        deleteElementContentsById("selected_country");
+        noCountriesLeftDOM();
+
+        return;
+      }
+
+      deleteElementContentsById("rand_country");
+      randCountryCodeStr = createRandCountryDOM(countries);
+    });
+
+    map.addListener('click', function(mapMouseEvent) {
+
+      if (countries.empty()) {
+        deleteElementContentsById("rand_country");
+        deleteAnswerDOM();
+        deleteElementContentsById("selected_country");
+        noCountriesLeftDOM();
+
+        return;
+      }
+
+      const geocoder = new google.maps.Geocoder;
+
+      geocoder.geocode({ 'location': mapMouseEvent.latLng }, function(address, status) {
+
+        if (status === "OK") {
+
+          deleteAnswerDOM();
+          deleteElementContentsById("selected_country");
+
+          const addressArray = Array.from(address);
+          const addressComps = addressArray[addressArray.length - 1].address_components
+
+          let countryCodeStr;
+          let countryNameStr;
+
+          for (let comp of addressComps) {
+
+            if (comp.short_name.length === 2) {
+              countryCodeStr = comp.short_name;
+              countryNameStr = comp.long_name;
+              break;
```

Comment:
I think it would be clearer to remove this `break` and make the following `if` an `else if`.

Reply:
Ok. I think I put the break just to save performance.